generators/client/templates/react/src/main/webapp/app/modules/account/activate/activate.tsx.ejs

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import { Row, Col, Alert } from 'reactstrap';
+import { Translate } from 'react-jhipster';
+import { parse } from 'query-string';
+
+import { activateAction, reset } from '../../../reducers/activate';
+
+const successAlert = (
+  <Alert color="success">
+    <Translate contentKey="activate.messages.success" />
+    <a className="alert-link">
+      <Translate contentKey="global.messages.info.authenticated.link" />
+    </a>.
+  </Alert>
+);
+
+const failureAlert = (
+  <Alert color="danger">
+    <Translate contentKey="activate.messages.error" />
+  </Alert>
+);
+
+export interface IActivateProps {
+  activateAction: Function;
+  reset: Function;
+  activationSuccess: boolean;
+  activationFailure: boolean;
+}
+
+export class ActivatePage extends React.Component<IActivateProps> {
+  getKeyFromLocation = () => {
+    const location = this.props['location'];
```

Comment:
deconstruct plz `const { location } = this.props;`
Anyway you should use match instead of location as its the correct way https://reacttraining.com/react-router/web/api/match
then you can use match.params.key